### PR TITLE
Send Shutdown event to event stream

### LINKED ISSUE #2153: Emit an event when node is shutting down
The event stream server should emit a new event `Shutdown` via every endpoint when it is about to shut down.

### PR DESCRIPTION
**Pull request by darthsiroftardis**
CHANGELOG: 

- Added a new `SseData` event called `Shutdown` that will emit an event to every `event-stream-server` endpoint when the node itself shutsdown
- Expanded the current SSE tests to include this new variant

Closes #2153

### CODE REVIEW

File: node/src/components/event_stream_server/sse_server.rs [hunk 1]
```diff
@@ -678,14 +675,16 @@ mod tests {
...
         should_not_filter_out(&fault, &MAIN_FILTER[..], getter.clone()).await;
         should_not_filter_out(&step, &MAIN_FILTER[..], getter.clone()).await;
+        should_not_filter_out(&shutdown, &MAIN_FILTER, getter.clone()).await;
 
         should_filter_out(&deploy_accepted, &MAIN_FILTER[..], getter.clone()).await;
         should_filter_out(&finality_signature, &MAIN_FILTER[..], getter.clone()).await;
 
         // `EventFilter::DeployAccepted` should filter out everything except `ApiVersion`s and
-        // `DeployAccepted`s.
+        // `DeployAccepted`s and Shutdown's.
```
Comment (rafal-ch) [inline node/src/components/event_stream_server/sse_server.rs]:
```
        // `EventFilter::DeployAccepted` should filter out everything except `ApiVersion`s,
        // `DeployAccepted`s and `Shutdown`s.
```

File: node/src/components/event_stream_server/tests.rs [hunk 2]
```diff
@@ -1171,13 +1171,19 @@ async fn should_limit_concurrent_subscribers() {
 /// `resources/test/sse_data_schema.json` across different versions of the codebase.
 #[test]
 fn schema() {
+    // To generate the contents to replace the input JSON files, run the test
+    // and print the `actual_schema by uncommenting the `println!`
```
Comment (rafal-ch) [inline node/src/components/event_stream_server/tests.rs]:
```suggestion
    // and print the `actual_schema` by uncommenting the `println!`
```

**Review by rafal-ch — APPROVED** [review]
LGTM with two minor nits.

File: node/src/components/event_stream_server/tests.rs (hunk 2 above)
Comment (darthsiroftardis) [reply node/src/components/event_stream_server/tests.rs]: Applied suggestion

File: node/src/components/event_stream_server/sse_server.rs (hunk 1 above)
Comment (darthsiroftardis) [reply node/src/components/event_stream_server/sse_server.rs]: Applied suggestion

File: node/src/components/event_stream_server/tests.rs [hunk 3]
```diff
@@ -1171,13 +1171,19 @@ async fn should_limit_concurrent_subscribers() {
...
+    // towards the end of the test.
     let schema_path = format!(
         "{}/../resources/test/sse_data_schema.json",
         env!("CARGO_MANIFEST_DIR")
     );
     let expected_schema = fs::read_to_string(schema_path).unwrap();
     let schema = schema_for!(SseData);
     let actual_schema = serde_json::to_string_pretty(&schema).unwrap();
+
+    // println!("{}", actual_schema);
```
Comment (nilz3ro) [inline node/src/components/event_stream_server/tests.rs:1200]: Do we need this?

Comment (darthsiroftardis) [reply node/src/components/event_stream_server/tests.rs:1200]: Yes, its for people to uncomment, when they change the schema and have to backfill the schema to account for the change

Comment (goral09) [reply node/src/components/event_stream_server/tests.rs:1200]: Then maybe add a comment explaining that :) I would remove that comment if I had seen it :)

Comment (rafal-ch) [reply node/src/components/event_stream_server/tests.rs:1200]: There is a comment explaining that just above, at the beginning of the test.

Comment (goral09) [reply node/src/components/event_stream_server/tests.rs:1200]: My brain would not go to the beginning of the test in search of explanation why a commented out `println!()` should not be removed.

**Review by Fraser999 — CHANGES_REQUESTED** [review]
I can approve after https://github.com/darthsiroftardis/casper-node/pull/15 is merged to this branch.  This currently has an issue whereby a shutdown event will be given the same ID as the first event emitted after re-starting the server.

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**